### ISSUE TITLE
Blex Mono NF (Windows Compatible) appears to have broken powerline glyphs

### ISSUE THREAD

**Issue by igniscyan** [ryanoasis/nerd-fonts#1122]
#### 🗹 Requirements

<!-- ⚠️ !!!
If you are unsure what you are reporting is an issue or
it's a general question please go to the gitter chat instead: https://gitter.im/ryanoasis/nerd-fonts
⚠️ !!! -->


<!-- ⚠️ !!! Issues not filled out with the template will be closed straight away and will only be responded to once filled properly ⚠️ !!! -->
<!-- By posting an issue you acknowledge the following: -->

- [x] I have searched the [issues](https://github.com/ryanoasis/nerd-fonts/issues) for my issue and found nothing related and/or helpful
- [x] I have searched the [FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting) for help
- [x] I have searched the [Wiki](https://github.com/ryanoasis/nerd-fonts/wiki) for help


#### 🎯 Subject of the issue

Experienced behavior:
<!-- Describe issue here -->

Glyphs corresponding to my shell theme frame appear to be broken, showing 3 separate segments.
Expected behavior: Glyph should be a single, connected line. Have confirmed it works correctly on other NerdFonts
<!-- Please do not forget to explain what you would have expected to see -->

Example symbols:
<!-- The codepoint of one or more affected symbols, or paste the text containing the issue here (no screenshot) -->
╭─ ~                                                                                                                                                                                                                         ✔  12:10:07
╰─

#### 🔧 Your Setup

- _Which font are you using (e.g. `Anonymice Powerline Nerd Font Complete.ttf`)?_
   - _Please give the full filename_ Blex Mono M NF 
- _Where did you get the file from (download link, self patched, source downloaded from link...)_
   - <!-- Type your answer here --> Download Link
- _Which terminal emulator are you using (e.g. `iterm2`, `urxvt`, `gnome`, `konsole`)?_
   - <!-- Type your answer here --> Windows Terminal (Preview, have tried AtlasRendering both on and off)
- _Are you using OS X, Linux or Windows? And which specific version or distribution?_
   - <!-- Type your answer here --> Windows 11, 22h2

#### ★ Screenshots (Optional)

<!-- Provide screenshots where appropriate, they help tremendously -->
![image](https://user-images.githubusercontent.com/26984470/218325988-9851d2dc-0c04-45ac-8c6f-6092431758b9.png)
<!-- Try to include the title of terminal windows and possibly settings dialogues -->
![image](https://user-images.githubusercontent.com/26984470/218326168-bf853ce8-a426-4b8d-97e5-f8f19cf8e241.png)![image](https://user-images.githubusercontent.com/26984470/218326179-089eb5f8-1efc-40c6-906e-b7113b43114e.png)

**Comment by Finii** [comment]
The glyphs in question are `256d` (`╭`) `2500` (`─`)

* These glyphs are not in the set that Nerd Font patches in.
* These glyphs are not in the source font

So they are also not available in the patched font:

![image](https://user-images.githubusercontent.com/16012374/218335097-3bb205f4-cef9-4ffb-af14-6ba715d5ed45.png)

Your client search for any font that has the glyphs and borrows them there (font fallback). That makes sense for letters and ordinary symbols but not for stuff that has to match exactly your line height and advance width.

**Comment by Finii** [comment]
Oh, I just realized that you are on `Windows Terminal`.
I must admit I have no clue how font fallback works there. Obviously it does in some way.
On Linux I know how to ask the system which font is used for fallback, but on Windows? :thinking: 
My Windows days are long past, unfortunately. And it was Windows that sparked my interest in fonts, as it came out with 3.1 and the new shiny TrueType fonts :grin: 

> have tried AtlasRendering both on and off

As you mention that, I know that there are these two rendering engines, but I have no clue how to switch. Maybe because I start the Windows box only up to test issues ppl report here :grimacing: Can you give me a hint where the switch is?

**Comment by igniscyan** [comment]
> Oh, I just realized that you are on `Windows Terminal`.
> 
> I must admit I have no clue how font fallback works there. Obviously it does in some way.
> 
> On Linux I know how to ask the system which font is used for fallback, but on Windows? :thinking: 
> 
> My Windows days are long past, unfortunately. And it was Windows that sparked my interest in fonts, as it came out with 3.1 and the new shiny TrueType fonts :grin: [1]
> 
> 
> 
> > have tried AtlasRendering both on and off
> 
> 
> 
> As you mention that, I know that there are these two rendering engines, but I have no clue how to switch. Maybe because I start the Windows box only up to test issues ppl report here :grimacing: Can you give me a hint where the switch is?
> 
> 
> 
> [1]
> 
> Well, that raised of course new problems, because the printer (I mean the company that operated the offset printer machines) took everything only in postscript, if they took files at all.

Hey Finii, thank you for the information! Unfortunately, it looks like Windows term has a default font fallback that can't be changed (there's a few issues on the windows term repo discussing it with quite a bit of activity, and currently someone is assigned to maybe add it?). 

As for the rendering engine, it's a feature that's enabled for me on the Windows 11 preview path, and I believe Windows Term (preview version in the MS store) lets you enable it with a dedicated rendering tab. In general I think Atlas is better, producing much clearer text than the solution they had before which distorted a lot of my favorite fonts. 

I've been itching to dualboot Linux and make it my primary spot for getting work done... my reasons for sticking with Windows these days are dwindling, and now that I'm back to using a Linux term on the daily... I may as well get a first class experience while I do it, lol. 

I ended up switching to a different font that had the correct characters, which is a bit of a bummer because I do love IBM plex mono. I'll have to look into finding a way to patch those ligatures(?) for it. 


Thank you again for the reply! Much appreciated.

**Comment by Finii** [comment]
> I ended up switching to a different font that had the correct characters, which is a bit of a bummer because I do love IBM plex mono. I'll have to look into finding a way to patch those ligatures(?) for it.

If you self patch it is relatively easy; 'Nerd Fonts has the tools you need' ;-)

* Well, get a shallow clone of the repo is maybe easiest, but takes time and space.
* Edit `bin/scripts/generate-extraglyphs.py` (see below)
* Run the script (needs to be run directly from `bin/scripts/`
* Use the `font-patcher` normally, either via ordinary call, or via `patch-em-all` if you want to patch one of the source fonts


Changes needed in `generate-extraglyphs.py`:
```python
font.selection.select(
    0x276C, # medium left-pointing angle bracket
    0x276D, # medium right-pointing angle bracket
    0x276E, # heavy left-pointing angle quotation mark
    0x276F, # heavy right-pointing angle quotation mark
    0x2770, # heavy left-pointing angle bracket
    0x2771, # heavy right-pointing angle bracket
)
```
Just add the codepoints you miss. They will be copied out of hack and used for any subsequent patching.

How to run the stuff:

**Comment by Finii** [comment]
I have seen other people also using these curvy line glyphs in their prompt. Is that some well known project or something?
Of course we could add them (easily, see above), if more people depend on them.

**Comment by igniscyan** [comment]
> I have seen other people also using these curvy line glyphs in their prompt. Is that some well known project or something? Of course we could add them (easily, see above), if more people depend on them.

It's a terminal prompt that has quite a large user base for zsh (and there are equivalents for bash and fish) called [Powerlevel10K](https://github.com/romkatv/powerlevel10k) It just gives you some nice context about the dir you're in/git status and presents it nicely. A lot of configs that use the multiline layout will opt in to using the characters I mentioned as a frame to visually connect prompt to status. 

In the mean time, I'll patch it myself using the instructions above - seems simple enough :) May be cool to see if the Nerdfonts user base finds those characters to be a useful inclusion, it may be too small of a niche.

**Comment by Finii** [comment]
Will be fixed by #1125

**Comment by igniscyan** [comment]
> Will be fixed by #1125

Thank you for looking into this!
